notebooks/article_1.py

```diff
@@ -0,0 +1,1170 @@
+# -  **Normalization** and **standardization**: These techniques adjust data to a common scale, making it easier to compare them.
+# -  **Encoding categorical data**: Transforming categorical variables into numerical formats.
+# 
+
+# ### Handle Irrelevant Features
+
+# An irrelevant feature is a data attribute or variable that doesn't offer useful or meaningful information for the task. Here are some reasons why removing them is beneficial:
+# 
+# - Eliminating irrelevant variables simplifies the analysis process.
+# - It ensures focus on meaningful factors by reducing unnecessary information and improving clarity in data exploration.
+# 
+# To continue, we'll remove the following variables and explain why for each.
+
+# In[70]:
+
+
+data_df = data_df.drop([
+    'Unnamed: 0',   # Irrelevant data
+    'latitude',     # Irrelevant data
+    'longitude',    # Irrelevant data
+    'zip_code',     # Irrelevant data
+    'id',           # Irrelevant data
+    'Unnamed: 6',   # Irrelevant data
+    'name',         # Irrelevant data
+    'labels',       # Same data as status
+    'state_code.1', # Same data as state_code
+    'is_CA',        # Binary column, not needed 
+    'is_NY',        # Binary column, not needed
+    'is_MA',        # Binary column, not needed
+    'is_TX',        # Binary column, not needed
+    'is_otherstate',# Binary column, not needed
+    'is_software',  # Binary column, not needed
+    'is_web',       # Binary column, not needed
+    'is_mobile',    # Binary column, not needed
+    'is_enterprise',# Binary column, not needed
+    'is_advertising',# Binary column, not needed
+    'is_gamesvideo',# Binary column, not needed
+    'is_ecommerce', # Binary column, not needed
+    'is_biotech',   # Binary column, not needed
+    'is_consulting',# Binary column, not needed
+    'is_othercategory',# Binary column, not needed
+    'object_id'],   # Irrelevant data
+    axis=1,
+).copy()
+data_df.head(2)
+
+
+# We've simplified the DataFrame. From 50 columns initially, it now contains 24 columns. Let's examine the data.
+
+# In[71]:
+
+
+data_df.info()
+
+
+# A preliminary review allows us to understand the nature of the data, especially focusing on identifying key data types that will help us to make decisions.
+# After discarding irrelevant variables we continue handle the Missing values.
+
+#  ### Missing Values
+
+# Missing values (NA) are data points in a dataset where information is simply not available or recorded for a specific attribute or variable. 
+# 
+# Importance of **handling** missing values:
+# 
+# - Preserve the integrity of your dataset. Without proper handling, NA can lead to biased or misleading results in statistical analyses. 
+#  
+# - Maintain the reliability of your analyses, making them more robust and trustworthy.
+
+# In[72]:
+
+
+x= data_df.isnull().sum()
+x.sort_values(ascending=False)
```

Comment:
In the output, can you hide those without missing values?

Reply:
missing_values = x[x > 0].sort_values(ascending=False) added